pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/WorkerConfig.java

```diff
@@ -432,6 +432,11 @@ public boolean getTlsEnabled() {
             + " to the init method of the secretproviderconfigurator"
     )
     private Map<String, String> secretsProviderConfiguratorConfig;
+    @FieldContext(
+            category = CATEGORY_FUNC_RUNTIME_MNG,
+            doc = "A set of the minimum amount of resources functions must request.  Support for this depends on function runtime."
+    )
+    private Resources functionInstanceMinResources;
```

Comment:
how is BC handled in this pull request? If function worker config doesn't have any settings, is this field null? if so, KubernetesRuntimeFactory.java will get NPE when accessing this field. 

Besides that, can we add a test case for covering this case here?

Reply:
@sijie the logic is handled in "validateMinResourcesRequired" please take a closer look.

```
        if (functionInstanceMinResources != null) {

```

Reply:
That test case is also covered in the unit test:
https://github.com/apache/pulsar/pull/3536/files#diff-da0caf611f277004fd188d0fc99e3178R179

Reply:
Ah I see. I misread the logic in KuberneteRuntimeFactory.

Can you also a test case that loading function worker config that doesn't contain those fields? Because that would cover people upgrading function worker from old version to new version but doesn't change the configuration file.

Also can you add the setting to `conf/functions_worker.yml` and document it?

Reply:
@sijie the unit test ready covers the case when the new worker config is not set:

https://github.com/apache/pulsar/pull/3536/files#diff-da0caf611f277004fd188d0fc99e3178R179

I will add the config to conf/functions_worker.yml

Reply:
@sijie I have added the config to conf/functions_worker.yml

Reply:
@jerrypeng - regarding the test, I mean loading the config from a file, because that's how people upgrade a function worker from an older version from a newer version. the test you pointed out is passing `null`, which doesn't actually cover the case I described. 

but anyway, I will let you decide whether such a test case is required or not.